sklearn/decomposition/tests/test_sparse_pca.py

```diff
@@ -59,14 +60,67 @@ def test_fit_transform():
     spca_lars = SparsePCA(n_components=3, method='lars', alpha=alpha,
                           random_state=0)
     spca_lars.fit(Y)
+    # variance computed by default shape (n_components, )
+    assert (spca_lars.explained_variance_.shape == (3,))
 
     # Test that CD gives similar results
     spca_lasso = SparsePCA(n_components=3, method='cd', random_state=0,
                            alpha=alpha)
     spca_lasso.fit(Y)
+    # variance computed by default shape (n_components, )
+    assert (spca_lasso.explained_variance_.shape == (3, ))
     assert_array_almost_equal(spca_lasso.components_, spca_lars.components_)
 
 
+def test_fit_transform_variance():
+    # This function asserts that the variance computed by SparsePCA is the
+    # same as the variance in PCA when the components are orthogonal.
+    alpha = 1
+    rng = np.random.RandomState(0)
+    X, _, _ = generate_toy_data(3, 10, (8, 8), random_state=rng)  # wide array
+    # init spca and pca
+    spca_lars = SparsePCA(n_components=3, method='lars', alpha=alpha,
+                          random_state=0)
+    spca_lars.fit(X)
+    assert spca_lars.explained_variance_.shape == (3,)
+    pca = PCA(n_components=3, random_state=0)
+
+    # fit PCA
+    pca.fit(X)
+
+    explained_variance = pca.explained_variance_
+
+    # force the components in spca_lars to be the same as in pca
+    spca_lars.components_ = pca.components_
+    # compute using private method
+    X = X - X.mean(axis=0)
+    explained_variance_sparse = _get_explained_variance(
+        X, spca_lars.components_)
```

Comment:
Maybe we can use `spca_lars.explained_variance_` here? We want to check that attribute, not so much the private function (since the private function is already check in another test).

Reply:
We cannot, since we set the `spca_lars.components_` to `pca.components_` in l94. If we use the attribute the computation is done on the original components.

One way to fix this would be to make the attribute a `@property`, is this ok? I would rather do this than import a private function as well.

Reply:
Hmm, yes indeed. I'm not sure we should make it a property just for this test, particularly if the calculation is expensive. Maybe just add a comment indicating that we want to test `spca_lars.explained_variance_` but are not able to do it directly since it is normally set in fit together with `components_`.